src/PaginatedList.cs

```diff
@@ -0,0 +1,32 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.EntityFrameworkCore;
+
+namespace MvcMovie
+{
+    public class PaginatedList<T> : List<T>
+    {
+        public int PageIndex { get; private set; }
+        public int TotalPages { get; private set; }
+
+        public PaginatedList(List<T> items, int count, int pageIndex, int pageSize)
+        {
+            PageIndex = pageIndex;
+            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+            this.AddRange(items);
+        }
+
+        public bool HasPreviousPage { get { return (PageIndex > 1); } }
```

Comment:
We usually prefer this format `public bool HasPrevious => PageIndex > 1` . It's only for more legibility